Core.ApplicationServices/GDPR/Write/DataProcessingRegistrationWriteService.cs

```diff
@@ -110,7 +116,45 @@ public class DataProcessingRegistrationWriteService : IDataProcessingRegistratio
                 .Bind(registration => registration.WithOptionalUpdate(parameters.General, UpdateGeneralData))
                 .Bind(registration => registration.WithOptionalUpdate(parameters.SystemUsageUuids, UpdateSystemAssignments))
                 .Bind(registration => registration.WithOptionalUpdate(parameters.Oversight, UpdateOversightData))
-                .Bind(registration => registration.WithOptionalUpdate(parameters.Roles, UpdateRolesData));
+                .Bind(registration => registration.WithOptionalUpdate(parameters.Roles, UpdateRolesData))
+                .Bind(registration => registration.WithOptionalUpdate(parameters.ExternalReferences, PerformReferencesUpdate));
+        }
+
+        private Result<DataProcessingRegistration, OperationError> PerformReferencesUpdate(DataProcessingRegistration dpr, IEnumerable<UpdatedExternalReferenceProperties> externalReferences)
+        {
+            //Clear existing state
+            dpr.ClearMasterReference();
```

Comment:
Det her kode er nok tæt på at være næsten det samme som det i itsystemusagewriteservice.
Lad os få det ud i en generisk "UpdateExternalReferencesCommand" som tager de parametre der skal til for at udføre kommandoen konkret på en bestemt type (i det her tilfælde er det vist så simpelt som RefernceRootType såvel som hvilken "Delete metode" der skal kaldes (en lambda til kommandoen". Den skal så "newes" her eller injectes hvis vi nu laver det til en IExternalReferencesBatchUpdateService.

Lad os tale om det hvis du er i tvivl men det er i hvert fald oplagt at få generaliseret det her for det er stort set det samme.